modules/cli/src/main/java/org/apache/ignite/internal/cli/commands/questions/ConnectToClusterQuestion.java

```diff
@@ -97,22 +98,40 @@ private String sessionNodeUrl() {
     }
 
     /**
-     * Ask if the user really wants to connect if we are already connected and the URL is different.
+     * Ask if the user really wants to connect if we are already connected and the URL or authentication parameters are different.
      *
-     * @param nodeUrl Node URL.
-     * @return {@link FlowBuilder} instance which provides the node URL if we are not connected or connected to different URL or interrupts
-     *         if user's answer is negative.
+     * @param input Node URL and authentication parameters.
+     * @return {@link FlowBuilder} instance which provides the connect call input if we are not connected or user agreed to reconnect or
+     *         interrupts if user's answer is negative.
      */
-    public FlowBuilder<Void, String> askQuestionIfConnected(String nodeUrl) {
+    public FlowBuilder<Void, ConnectCallInput> askQuestionIfConnected(ConnectCallInput input) {
         SessionInfo sessionInfo = session.info();
-        if (sessionInfo != null && !Objects.equals(sessionInfo.nodeUrl(), nodeUrl)) {
-            QuestionUiComponent question = fromYesNoQuestion(
-                    "You are already connected to the %s, do you want to connect to the %s?",
-                    UiElements.url(sessionInfo.nodeUrl()), UiElements.url(nodeUrl)
-            );
-            return Flows.acceptQuestion(question, () -> nodeUrl);
+        if (sessionInfo != null) {
+            if (!Objects.equals(sessionInfo.nodeUrl(), input.url())) {
+                return Flows.acceptQuestion(fromYesNoQuestion(
+                        "You are already connected to the %s, do you want to connect to the %s?",
+                        UiElements.url(sessionInfo.nodeUrl()), UiElements.url(input.url())
+                ), () -> input);
+            }
+
+            String oldUsername = sessionInfo.username();
+            String newUsername = input.username();
+
+            if (newUsername != null) {
+                if (oldUsername != null) {
+                    return Flows.acceptQuestion(fromYesNoQuestion(
+                            "You are already connected to the %s as %s, do you want to connect as %s?",
+                            UiElements.url(sessionInfo.nodeUrl()), string(oldUsername), string(newUsername)
+                    ), () -> input);
+                } else {
+                    return Flows.acceptQuestion(fromYesNoQuestion(
+                            "You are already connected to the %s, do you want to connect as %s?",
+                            UiElements.url(sessionInfo.nodeUrl()), string(newUsername)
```

Comment:
I think it is better to introduce `username` UI element. In the future, we might change the way we display usernames.